### PR TITLE
Use mock JWT and JWKS from codebase for end-user auth tutorial

### PR DESCRIPTION
**Pull request by diemtvu**
This simplify setup for end-user authentication demo.

### CODE REVIEW

**Comment by geeknoid** [comment]
Please replace all occurrences of https://raw.githubusercontent.com with {{< github_file >}} as described here: https://preliminary.istio.io/about/contribute/writing-a-new-topic/#github

File: content/docs/tasks/security/authn-policy/index.md [hunk 1]
```diff
@@ -457,7 +457,7 @@ $ kubectl delete destinationrules httpbin -n bar
 
 ## End-user authentication
 
-To experiment with this feature, you will need a valid JWT (corresponding to the JWKS endpoint you want to use for the demo). In this tutorial, we will use a test [JWT](https://raw.githubusercontent.com/istio/istio/master/security/tools/jwt/samples/demo.jwt) and [JWKS endpoint](https://raw.githubusercontent.com/istio/istio/master/security/tools/jwt/samples/jwks.json) from Istio code base.
+To experiment with this feature, you will need a valid JWT (corresponding to the JWKS endpoint you want to use for the demo). In this tutorial, we will use a test [JWT]({{< github_file >}}/security/tools/jwt/samples/demo.jwt) and [JWKS endpoint]({{< github_file >}}/master/security/tools/jwt/samples/jwks.json) from Istio code base.
```
Comment (geeknoid) [inline content/docs/tasks/security/authn-policy/index.md]: Remove /master in the urls, this is part of what {{<github_file>}} fills in.

**Comment by geeknoid** [comment]
This looks good. But we'll need to wait until the corresponding PR is in istio/istio before pushing this change in.

File: content/docs/tasks/security/authn-policy/index.md [hunk 2]
```diff
@@ -552,12 +544,28 @@ $ curl $INGRESS_HOST/headers -s -o /dev/null -w "%{http_code}\n"
...
 
-{{< text bash >}}
+{{< text bash>}}
+$ TOKEN=$(curl {{< github_file >}}/security/tools/jwt/samples/demo.jwt -s)
 $ curl --header "Authorization: Bearer $TOKEN" $INGRESS_HOST/headers -s -o /dev/null -w "%{http_code}\n"
 200
 {{< /text >}}
 
-You may want to try to modify token or policy (e.g change issuer, audiences, expiry date etc) to observe other aspects of JWT validation.
+You may want to use the script [`gen-jwt.py`]({{< github_tree >}}/security/tools/jwt/samples) to generate new tokens for testing with different issuer, audiences, expiry date etc to observe other aspects of JWT validation. For example, the command below creates a token that will expire in 5 seconds. As you will see, requests using that token will be authenticated successfully at first, but then be rejected after 5 seconds:
```
Comment (rcaballeromx) [inline content/docs/tasks/security/authn-policy/index.md]: Watch out for the line length. My understanding is that 80 characters is the correct length for each line.

Comment (rcaballeromx) [reply content/docs/tasks/security/authn-policy/index.md]: Avoid using conjunctives and future tense. Writing only on the present tense helps readers identify with the content and gives the text a more active tone. Furthermore, it makes the text clearer and a lot easier to understand. I suggest these edits:
To observe other aspects of JWT validation, use the script [`gen-jwt.py`]({{< github_tree >}}/security/tools/jwt/samples) to generate new tokens to test with different issuer, audiences, expiry date, etc. For example, the command below creates a token that expires in 5 seconds. As you see, Istio authenticates requests using that token successfully at first but rejects them after 5 seconds:

File: content/docs/tasks/security/authn-policy/index.md [hunk 3]
```diff
@@ -457,15 +457,7 @@ $ kubectl delete destinationrules httpbin -n bar
...
-To experiment with this feature, you will need a valid JWT (corresponding to the JWKS endpoint you want to use for the demo). In this tutorial, we will use a
-test JWT signed by Google service account, but of course, you can also use your own JWT/JWKS endpoint for the demo.
-
-You can create a test JWT for your Google service account (assume it's stored in the environment variable `$SVC_ACCOUNT`) by following the instructions
-[here]({{< github_tree >}}/security/tools/jwt). Keep it in `$TOKEN`. Also set `$JWKS` corresponding to the service account `$SVC_ACCOUNT` as below:
-
-{{< text bash >}}
-$ JWKS=https://www.googleapis.com/service_accounts/v1/jwk/${SVC_ACCOUNT}
-{{< /text >}}
+To experiment with this feature, you will need a valid JWT (corresponding to the JWKS endpoint you want to use for the demo). In this tutorial, we will use a test [JWT]({{< github_file >}}/security/tools/jwt/samples/demo.jwt) and [JWKS endpoint]({{< github_file >}}/security/tools/jwt/samples/jwks.json) from Istio code base.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/authn-policy/index.md]: Again, watch for the line length. See my other comment regarding verb tenses. Additionally, consider adding articles before specific elements. Let the users know that it is "this" test the one they need and no other, the same goes for the endpoint. Consider this edit: 
To experiment with this feature, you need a valid JWT. The JWT must correspond to the JWKS endpoint you want to use for the demo. In this tutorial, we use this [JWT test]({{< github_file >}}/security/tools/jwt/samples/demo.jwt) and this [JWKS endpoint]({{< github_file >}}/security/tools/jwt/samples/jwks.json) from the Istio code base.

Comment (diemtvu) [reply content/docs/tasks/security/authn-policy/index.md]: Done.

File: content/docs/tasks/security/authn-policy/index.md (hunk 2 above)
Comment (diemtvu) [reply content/docs/tasks/security/authn-policy/index.md]: Done.

Btw, I use 160 for line limit. Do you want to change it to 80?

**Comment by geeknoid** [comment]
Line length guideline for Istio is 160.

**Pull request merged**